lib/puppet/application/lookup.rb

```diff
@@ -49,8 +57,44 @@ class Puppet::Application::Lookup < Puppet::Application
   def run_command
     options[:keys] = command_line.args
 
-    puts "Hello World!"
-    puts options
-  end
+    if options[:keys].empty?
+     raise "No keys were given to lookup."
+    end
+
+    if (options[:sort_merge_arrays] || options[:merge_hash_arrays] || options[:prefix] || options[:upack_arrays]) && options[:merge] != 'deep'
+      raise "The options #{DEEP_MERGE_OPTIONS} are only available with '--merge deep'\n#{RUNHELP}"
+    end
+
+    node = Puppet::Node.indirection.find("#{options[:node]}")
```

Comment:
Even when nil is passed in here we still seem to end up with a compiler and scope that work, at least in my environment. I was running on a master and it successfully found the data I had in the modules in my production environment. However I don't really know WHY this works or how... so I think in the mean time I'll just change this to error out if you don't specify a node.